### PR TITLE
clown items, henk

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
The clown has gone without any researchable tech for too long! Im adding a clown headset and clown voicebox, both researchable under Clown Tech.

update: the honk filter has been moved to the headset tgui, and you cant insert amp keys

## Why It's Good For The Game
henk honk

Makes it easier to use the "awesome" Comic Sans Purple text without having to risk your mic being stolen!

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

henk_floor

henk_use

image

</details>

## Changelog

:cl:
add: Added clown headset/voicebox
tweak: Added both to the Clown Tech node
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/tongue.dm [hunk 1]
```diff
@@ -255,6 +255,19 @@
 /obj/item/organ/tongue/robot/handle_speech(datum/source, list/speech_args)
 	speech_args[SPEECH_SPANS] |= SPAN_ROBOT
 
+/obj/item/organ/tongue/honk
+	name = "clown voicebox"
+	desc = "A voice synthesizer that makes you 45% funnier.
```
Comment (Sarchutar) [inline code/modules/surgery/organs/tongue.dm]:
```suggestion
	desc = "A voice synthesizer that makes you 45% funnier."
```

**Review by Sarchutar — DISMISSED** [review]
I'm not a maintainer but code seems fine to me and seems to work

File: code/game/objects/items/devices/radio/headset.dm [hunk 2]
```diff
@@ -290,6 +290,30 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 /obj/item/radio/headset/silicon/can_receive(freq, level)
 	return ..(freq, level, TRUE)
 
+/obj/item/radio/headset/clown
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/radio/headset.dm:294]: Does clown not already get one? I swearthey have a pink and yellow headset already. Might need to be renamed ro be clearly not the clowns starting headset or use a component based approach instead

Comment (etherware-novice) [reply code/game/objects/items/devices/radio/headset.dm:294]: image
nope, just the service radio

File: code/game/objects/items/devices/radio/headset.dm [hunk 3]
```diff
@@ -290,6 +290,30 @@ GLOBAL_LIST_INIT(channel_tokens, list(
...
+	desc = "A headset with a clowning filter, honk!"
+	icon_state = "honk_headset"
+	var/honken = TRUE
+
+/obj/item/radio/headset/clown/talk_into_impl(atom/movable/M, message, channel, list/spans, datum/language/language, list/message_mods)
+	if(honken)
+		spans |= SPAN_CLOWN
+	. = ..()
+
+/obj/item/radio/headset/clown/AltClick(mob/user)
```
Comment (itsmeow) [inline code/game/objects/items/devices/radio/headset.dm]: This should use the existing TGUI volume functionality instead of Alt+Click.

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/radio/headset.dm (hunk 3 above)
Comment (etherware-novice) [reply code/game/objects/items/devices/radio/headset.dm]: i have 0 knowledge of how to use tgui, youll have to tell me how

Comment (itsmeow) [reply code/game/objects/items/devices/radio/headset.dm]: you know how as command, AI, etc you can open the radio and check "Subspace Tx" or "High Volume Mode"

add a button for that, but for honk (high volume) mode.

look at the command radios for examples.

File: tgui/packages/tgui/interfaces/Radio.js [hunk 4]
```diff
@@ -86,6 +87,14 @@ export const Radio = (props, context) => {
                   content={`Subspace Tx ${subspace ? 'ON' : 'OFF'}`}
                   onClick={() => act('subspace')} />
               )}
+              {!!honk && (
+                <Button
+                  ml={1}
+                  icon="bullhorn"
+                  selected={useHonk}
+                  content={`${useHonk ? 'HONK!' : 'honk.'}`}
+                  onClick={() => act('honk')} />
```
Comment (itsmeow) [inline tgui/packages/tgui/interfaces/Radio.js:97]: `color={useHonk ? "pink" : null}`

File: code/game/objects/items/devices/radio/headset.dm [hunk 5]
```diff
@@ -294,32 +294,14 @@ GLOBAL_LIST_INIT(channel_tokens, list(
...
-	. = ..()
-	.["honk"] = honken
-
-/obj/item/radio/headset/clown/ui_act(action, params, datum/tgui/ui)
-	. = ..()
-	if(action == "honk")
-		honken = !honken
-		. = TRUE
+	honken = TRUE  // honk filter
+	use_honken = TRUE
```
Comment (itsmeow) [inline code/game/objects/items/devices/radio/headset.dm]: this should start off like other command headsets.

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/Radio.js (hunk 4 above)
Comment (itsmeow) [reply tgui/packages/tgui/interfaces/Radio.js:97]: also why is this wrapped in a template string? Just do `content={useHonk ? 'HONK!' : 'honk.'}`

File: code/game/objects/items/devices/radio/headset.dm (hunk 5 above)
Comment (etherware-novice) [reply code/game/objects/items/devices/radio/headset.dm]: oh are those disabled roundstart? i havent played a head in ages

File: tgui/packages/tgui/interfaces/Radio.js (hunk 4 above)
Comment (etherware-novice) [reply tgui/packages/tgui/interfaces/Radio.js:97]: which part is a template string?

Comment (etherware-novice) [reply tgui/packages/tgui/interfaces/Radio.js:97]: o

**Comment by itsmeow** [comment]
Is the ON state for the honk button pink?

If so, approved.

**Review by itsmeow — APPROVED** [review]

File: code/game/objects/items/devices/radio/headset.dm [hunk 6]
```diff
@@ -290,6 +290,22 @@ GLOBAL_LIST_INIT(channel_tokens, list(
...
+
+/obj/item/radio/headset/clown/talk_into_impl(atom/movable/M, message, channel, list/spans, datum/language/language, list/message_mods)
+	if(honken)
+		spans |= SPAN_CLOWN
+	. = ..()
+
+/obj/item/radio/headset/clown/examine(mob/user)
+	. = ..()
+	. += "The honkening filter is currently [honken ? "On" : "Off"], use Alt-Click to toggle."
```
Comment (itsmeow) [inline code/game/objects/items/devices/radio/headset.dm]: gah wait, fix this

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/radio/headset.dm (hunk 6 above)
Comment (itsmeow) [reply code/game/objects/items/devices/radio/headset.dm]: probably just remove

Comment (etherware-novice) [reply code/game/objects/items/devices/radio/headset.dm]: wait why? the command loudmode has a similar examine str

Comment (itsmeow) [reply code/game/objects/items/devices/radio/headset.dm]: `use Alt-Click to toggle.` this is not accurate, is it?

Comment (etherware-novice) [reply code/game/objects/items/devices/radio/headset.dm]: oh oops

**Review by itsmeow — APPROVED** [review]
Okay. Should actually be good now.

File: code/game/objects/items/devices/radio/radio.dm [hunk 7]
```diff
@@ -33,6 +33,9 @@
 	var/use_command = FALSE  // If true, broadcasts will be large and BOLD.
 	var/command = FALSE  // If true, use_command can be toggled at will.
 
+	var/use_honken = FALSE
+	var/honken = FALSE  // same as command voice but HONK
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/radio/radio.dm:37]: If this doesn't work on any headsets except the clown headset, then this should be entirely contained within the clown headset

File: code/modules/surgery/organs/tongue.dm [hunk 8]
```diff
@@ -255,6 +255,19 @@
...
+	desc = "A voice synthesizer that makes you 45% funnier."
+	slot = ORGAN_SLOT_TONGUE
+	icon_state = "tonguerobot"  // looks sufficiently tech-y enough to pass
+	say_mod = "honks"
+	attack_verb = list("honked", "clowns")
+	modifies_speech = TRUE
+	taste_sensitivity = 25 // not as good as an organic tongue
+
+/obj/item/organ/tongue/honk/handle_speech(datum/source, list/speech_args)
+	speech_args[SPEECH_SPANS] |= SPAN_CLOWN
```
Comment (PowerfulBacon) [inline code/modules/surgery/organs/tongue.dm:272]: Does this need to call the parent?

File: code/modules/surgery/organs/tongue.dm [hunk 9]
```diff
@@ -255,6 +255,19 @@
...
 
+/obj/item/organ/tongue/honk
+	name = "clown voicebox"
+	desc = "A voice synthesizer that makes you 45% funnier."
+	slot = ORGAN_SLOT_TONGUE
+	icon_state = "tonguerobot"  // looks sufficiently tech-y enough to pass
+	say_mod = "honks"
+	attack_verb = list("honked", "clowns")
+	modifies_speech = TRUE
+	taste_sensitivity = 25 // not as good as an organic tongue
```
Comment (PowerfulBacon) [inline code/modules/surgery/organs/tongue.dm:269]: Most of this stuff is copy and pasted off of the robot tongue, so make it a subtype of it

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/radio/radio.dm (hunk 7 above)
Comment (etherware-novice) [reply code/game/objects/items/devices/radio/radio.dm:37]: im futureproofing it incase i want to add a clown encryption key later

File: code/modules/surgery/organs/tongue.dm (hunk 9 above)
Comment (itsmeow) [reply code/modules/surgery/organs/tongue.dm:269]: Bad idea, don't do that - there's two fields from the robot tongue that are copied, but the robot tongue also adds a bunch of stuff like languages and emp reaction.

File: code/modules/surgery/organs/tongue.dm (hunk 8 above)
Comment (itsmeow) [reply code/modules/surgery/organs/tongue.dm:272]: No, robot tongue doesn't

File: code/game/objects/items/devices/radio/radio.dm [hunk 10]
```diff
@@ -219,6 +224,9 @@
 				else
 					recalculateChannels()
 				. = TRUE
+		if("honk")  // if i ever make a radio key for this it wont be broken :)
+			use_honken = !use_honken
```
Comment (itsmeow) [inline code/game/objects/items/devices/radio/radio.dm:232]:
```
if(!honken)
   return
```

We don't want people hrefing the ability to be the clown.

Comment (itsmeow) [reply code/game/objects/items/devices/radio/radio.dm:232]: Can you add the same to the command check above please?

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Review by PowerfulBacon — APPROVED** [review]
Code seems fine, not convinced if its worth the added headset code.

**Review by itsmeow — APPROVED** [review]

**Pull request merged**